src/main/java/seedu/clialgo/command/AddCodeCommand.java

```diff
@@ -45,4 +47,17 @@ public void execute(TopicManager topicManager, Ui ui, FileManager fileManager, B
             new InvalidCommand().execute(topicManager, ui, fileManager, buffer);
         }
     }
+
+    /**
+     * An overridden method that checks for equality of <code>AddCodeCommand</code> objects.
+     *
+     * @param otherCommand The other <code>AddCodeCommand</code> object to be checked against.
+     * @return A boolean value to determine whether the <code>AddCodeCommand</code> objects are equal.
+     */
+    @Override
+    public boolean equals(Command otherCommand) {
+        AddCodeCommand otherAddCodeCommand = (AddCodeCommand) otherCommand;
+        return Objects.equals(this.name, otherAddCodeCommand.name);
```

Comment:
To check for equality of AddCodeCommand, do you also need to check whether the topic and importance are the same as well?

Reply:
im not too sure, what do you suggest?

Reply:
I agree with @ong-ck

Reply:
Maybe you could consider this

```suggestion
        public boolean equals(Command otherCommand) {
            AddCodeCommand otherAddCodeCommand = (AddCodeCommand) otherCommand;
            boolean isNameEqual = Objects.equals(this.name, otherAddCodeCommand.name);
            boolean isTopicEqual = Objects.equals(this.topic, otherAddCodeCommand.topic);
            boolean isImportanceEqual = this.importance == otherAddCodeCommand.importance;
            return isNameEqual && isTopicEqual  && isImportanceEqual;
        }
```

Reply:
okay, will make changes

Reply:
is my current one okay? i combined 3 of them into the return statement

Reply:
ohh right